master/internal/api_experiment.go

```diff
@@ -2100,11 +2100,16 @@ func sortExperiments(sortString *string, experimentQuery *bun.SelectQuery) error
 				"."+metricQualifier)
 			if !slices.Contains(SummaryMetricStatistics, metricQualifier) {
 				return status.Errorf(codes.InvalidArgument,
-					"sort training metrics by statistic: count, last, max, min, or sum")
+					"sort training metrics by statistic: last, max, min, or mean")
+			}
+			if metricQualifier == "mean" { //nolint: goconst
+				locator := bun.Safe(fmt.Sprintf("trials.summary_metrics->'avg_metrics'->'%s'", metricName))
```

Comment:
issue: I am worried about the bun.Safe + Sprintf usage in this method. it seems `metricName` comes from the user. let's say it contains `'` symbol.
`.ColumnExpr("trials.summary_metrics->'avg_metrics'->?", metricName)` would have escaped it, but here it's wrapped in `bun.Safe`, so there might be a risk of an injection?

Reply:
Looking at how we do metric queries in postgres_trials, it looks like we'd want to use the bun subqueries?  `subq.ColumnExpr("trials.summary_metrics->'avg_metrics'->>?", metricName)`

I agree better not to use bun.Safe

Reply:
will do!